packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -657,11 +727,26 @@ class FuchsiaDevice extends Device {
     return ports;
   }
 
+  /// Determine if the fuchsia device is running a workstation build.
+  ///
+  /// If the device is running a workstation build, `session_control` should be
+  /// used to launch apps, otherwise `tiles_ctl` should be used.
+  Future<bool> isWorkstation() async {
```

Comment:
Please don't hardcode any product configuration details into the flutter tool because it should be able to work with any product configuration.

Reply:
You are right. I am thinking of an alternative way to detect whether we should use tiles or session. Maybe detecting the existence of a session_control binary would work?